shark-hprof/src/main/java/shark/StreamingRecordReaderAdapter.kt

```diff
@@ -67,148 +67,148 @@ class StreamingRecordReaderAdapter(private val streamingHprofReader: StreamingHp
         STRING_IN_UTF8 -> {
           val recordPosition = reader.bytesRead
           val record = reader.readStringRecord(length)
-          listener.onHprofRecord(recordPosition, record)
+          listener.onHprofRecord(recordPosition, length, record)
         }
         LOAD_CLASS -> {
           val recordPosition = reader.bytesRead
           val record = reader.readLoadClassRecord()
-          listener.onHprofRecord(recordPosition, record)
+          listener.onHprofRecord(recordPosition, length, record)
         }
         STACK_FRAME -> {
           val recordPosition = reader.bytesRead
           val record = reader.readStackFrameRecord()
-          listener.onHprofRecord(recordPosition, record)
+          listener.onHprofRecord(recordPosition, length, record)
         }
         STACK_TRACE -> {
           val recordPosition = reader.bytesRead
           val record = reader.readStackTraceRecord()
-          listener.onHprofRecord(recordPosition, record)
+          listener.onHprofRecord(recordPosition, length, record)
         }
         ROOT_UNKNOWN -> {
           val recordPosition = reader.bytesRead
           val record = reader.readUnknownGcRootRecord()
-          listener.onHprofRecord(recordPosition, GcRootRecord(record))
+          listener.onHprofRecord(recordPosition, reader.bytesRead - recordPosition, GcRootRecord(record))
```

Comment:
This likely points to a miss in the `OnHprofRecordTagListener` interface where it's unclear when you'll get -1 or not (ie if the length is known or not, in the heap segment case)